File: backend/src/backend/api/hsl.clj
(ns backend.api.hsl
  (:require [clj-http.client :as client]))

(def url "https://api.digitransit.fi/routing/v1/routers/hsl/index/graphql")

(defn get-hsl-nodes-in-radius
  [lat lon radius]
  (let [query    (str "{stopsByRadius (lat: " lat ", lon: " lon ", radius: " radius ") {
                          edges {
                            node {
                              distance
                                stop {
                                  name
                                  lat
                                  lon
                                  wheelchairBoarding}}}}}")
        response (client/post url {:content-type "application/graphql"
                                   :body         query
                                   :as           :json})]
    (map :node (-> response
                   :body
                   :data
                   :stopsByRadius
                   :edges))))

File: frontend/src/api/hsl.ts


export async function getStopsInRadius(lat: number, lng: number, radius: number) {
    return {
        "stops": [
            {
                "name": "Pasilan asema",
                "lat": 60.1981,
                "lon": 24.93365,
                "distance": 225,
                "accessible": true
            },
            {
                "name": "Pasilan asema",
                "lat": 60.198075,
                "lon": 24.933694,
                "distance": 261,
                "accessible": false
            },
            {
                "name": "Pasilan asema",
                "lat": 60.198002,
                "lon": 24.933908,
                "distance": 294,
                "accessible": false
            }
        ]
    }
}